### PR TITLE
Fix a bug that piped was handling only one stage LogPersister at a time

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/piped/logpersister/persister.go [hunk 1]
```diff
@@ -106,56 +108,67 @@ L:
...
 		sp := v.(*stageLogPersister)
+
 		if sp.isStale(p.stalePeriod) {
 			completedKeys = append(completedKeys, sp.key)
-		} else {
-			sp.flush(ctx)
+			deletes++
+			return true
 		}
-		return false
```
Comment (nghialv) [inline pkg/app/piped/logpersister/persister.go:142]: This causes the problem.

File: pkg/app/piped/logpersister/persister.go [hunk 2]
```diff
@@ -106,56 +108,67 @@ L:
...
+func (p *persister) flushAll(ctx context.Context) int {
+	var num = 0
+
 	p.stagePersisters.Range(func(_, v interface{}) bool {
 		sp := v.(*stageLogPersister)
 		if !sp.isStale(p.stalePeriod) {
+			num++
 			go sp.flushFromLastCheckpoint(ctx)
 		}
-		return false
```
Comment (nghialv) [inline pkg/app/piped/logpersister/persister.go:157]: This causes the problem.

File: pkg/app/piped/logpersister/persister.go (hunk 1 above)
Comment (nakabonne) [reply pkg/app/piped/logpersister/persister.go:142]: Turned out 👍

**Pull request merged**